test/optimizer/rewriter_test.cpp

```diff
@@ -0,0 +1,425 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// operator_test.cpp
+//
+// Identification: test/optimizer/operator_test.cpp
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include <algorithm>
+#include "common/harness.h"
+
+#include "optimizer/operators.h"
+#include "optimizer/rewriter.h"
+#include "expression/constant_value_expression.h"
+#include "expression/comparison_expression.h"
+#include "expression/tuple_value_expression.h"
+#include "expression/operator_expression.h"
+#include "type/value_factory.h"
+#include "type/value_peeker.h"
+#include "optimizer/rule_rewrite.h"
+
+namespace peloton {
+
+namespace test {
+
+using namespace optimizer;
+
+class RewriterTests : public PelotonTest {};
+
+TEST_F(RewriterTests, SingleCompareEqualRewritePassFalse) {
+  type::Value leftValue = type::ValueFactory::GetIntegerValue(3);
+  type::Value rightValue = type::ValueFactory::GetIntegerValue(2);
+  auto left = new expression::ConstantValueExpression(leftValue);
+  auto right = new expression::ConstantValueExpression(rightValue);
+  auto common = new expression::ComparisonExpression(ExpressionType::COMPARE_EQUAL, left, right);
+
+  Rewriter *rewriter = new Rewriter();
+  auto rewrote = rewriter->RewriteExpression(common);
+
+  delete rewriter;
```

Comment:
The convention that I think we usually use is to use smart pointers with a custom scope to achieve the same thing.